### PR TITLE
Translations bugfix and update

### PR DESCRIPTION
**Pull request by VlasovVitaly**
Bug fixed
u4eelfd
Ammunition type was not translated.

Improved code readability in this piece of file.

Added lots of helping comments for translators.

### CODE REVIEW

File: src/item.cpp [hunk 1]
```diff
@@ -1076,34 +1078,82 @@ std::string item::info(bool showtext, std::vector<iteminfo> &dump_ref) const
         it_tool* tool = dynamic_cast<it_tool*>(type);
 
         if ((tool->max_charges)!=0) {
-            std::string charges_line = _("Charges"); //;
+            int t_max;
+            const std::string t_ammo_name = _(ammo_name(tool->ammo_id).c_str());
```
Comment (BevapDin) [inline src/item.cpp]: `ammo_name` should probably do the translation. I see at least 3 further calls to it where the result is not translated:

src/game.cpp:
```C++
add_msg(m_info, _("Out of %s!"), ammo_name(tool->ammo_id).c_str());
```

src/npctalk.cpp:
```C++
phrase.replace(fa, l, ammo_name(me->weapon.ammo_type()) );
```

src/player.cpp:
```C++
add_msg(m_info, _("That %s cannot be used on a %s."), used->tname().c_str(),
                       ammo_name(guntype->ammo).c_str());
```

That would also be kind of consistent with how `item::tname`, `monster::name`, etc. return a translated name.

File: src/item.cpp [hunk 2]
```diff
@@ -1076,34 +1078,82 @@ std::string item::info(bool showtext, std::vector<iteminfo> &dump_ref) const
...
+            std::string temp_fmt;
+            std::string charges_line = _("Charges");
             dump->push_back(iteminfo("TOOL",charges_line+ ": " + to_string(charges)));
```
Comment (BevapDin) [inline src/item.cpp]: You could also use `string_format` directly here and get rid of the `charges_line` variable:
```C++
string_format( _("Charges: %d"), charges)
```

File: src/item.cpp [hunk 3]
```diff
@@ -1076,34 +1078,82 @@ std::string item::info(bool showtext, std::vector<iteminfo> &dump_ref) const
         it_tool* tool = dynamic_cast<it_tool*>(type);
 
         if ((tool->max_charges)!=0) {
-            std::string charges_line = _("Charges"); //;
+            int t_max;
```
Comment (BevapDin) [inline src/item.cpp]: Declaring the variables `t_max`, `temp_fmt` etc. outside of the if-blocks where they are used makes the impressions there is some connection between the logic in the block and the code afterwards. I expected something like this:
```C++
int f_max;
// set f_max depending on various conditions:
if( condition1 ) {
    f_max = value1;
} else {
    f_max = value2;
}
print( "max: %d", f_max );
```

But here, there is not such connection, each block uses the variable separately. Therefor the variables should be declared in each block (and you can actually make them const there):
```C++
if( condition1 ) {
    const int f_max = value1;
    print( "max: %d", f_max );
} else {
    const int f_max = value2;
    print( "max: %d", f_max );
}
```

Or maybe just move the `dump->push_back(iteminfo(` line out of the blocks and at the end as it is common to all blocks.

File: src/item.cpp (hunk 1 above)
Comment (VlasovVitaly) [inline src/item.cpp]: I need to check these cases.

File: src/item.cpp (hunk 2 above)
Comment (VlasovVitaly) [inline src/item.cpp]: Yes. Good.

File: src/item.cpp (hunk 3 above)
Comment (VlasovVitaly) [inline src/item.cpp]: Both variants has strengths and weaknesses. I like first one, but this is not obvious behaviour. Second is obvious, but require many lines in every logic block...

Guess, i will choose moving call of dump->push_back() to end of logic. This is not so strange.

**Pull request merged**